subsys/net/lib/lwm2m/lwm2m_rw_senml_json.c

```diff
@@ -17,262 +17,249 @@ LOG_MODULE_REGISTER(LOG_MODULE_NAME);
-	/* flags */
-	uint8_t writer_flags;
-
-	/* base name */
-	struct lwm2m_obj_path base_name;
-	/* Add Base name */
-	bool add_base_name_to_start;
+struct senml_string_bn_payload {
+	const char *base_name;
+	const char *name;
+	const char *val_string;
 };
 
-struct json_in_formatter_data {
-	/* name info */
-	uint16_t name_offset;
-	uint16_t name_len;
-
-	/* value info */
-	uint16_t value_offset;
-	/* Value length */
-	uint16_t value_len;
-
-	/* state */
-	uint16_t offset;
+struct senml_string_payload {
+	const char *name;
+	const char *val_string;
+};
 
-	/* flags */
-	uint8_t json_flags;
+struct senml_opaque_payload {
+	const char *name;
+	struct json_obj_token val_opaque;
 };
 
-/* some temporary buffer space for format conversions */
-static char json_buffer[TOKEN_BUF_LEN];
+struct senml_opaque_bn_payload {
+	const char *base_name;
+	const char *name;
+	struct json_obj_token val_opaque;
+};
 
-static void json_add_char(struct lwm2m_input_context *in, struct json_in_formatter_data *fd)
-{
-	if ((fd->json_flags & T_VALUE) ||
-	    ((fd->json_flags & T_STRING_BEGIN) && !(fd->json_flags & T_STRING_END))) {
-		if (fd->json_flags & T_VALUE) {
-			fd->value_len++;
-			if (fd->value_len == 1U) {
-				fd->value_offset = fd->offset;
-			}
-		} else {
-			fd->name_len++;
-			if (fd->name_len == 1U) {
-				fd->name_offset = fd->offset;
-			}
-		}
-	}
-}
+struct senml_boolean_payload {
+	const char *name;
+	bool val_bool;
+};
 
-static struct lwm2m_senml_json_context *seml_json_context_get(struct lwm2m_block_context *block_ctx)
-{
-	if (block_ctx) {
-		return &block_ctx->senml_json_ctx;
-	}
+struct senml_boolean_bn_payload {
+	const char *base_name;
+	const char *name;
+	bool val_bool;
+};
 
-	return NULL;
-}
+struct senml_float_payload {
+	const char *name;
+	struct json_obj_token val_float;
+};
 
-static int json_atribute_decode(struct lwm2m_input_context *in, struct json_in_formatter_data *fd)
-{
-	uint8_t attrbute_name[3];
+struct senml_float_bn_payload {
+	const char *base_name;
+	const char *name;
+	struct json_obj_token val_float;
+};
 
-	if (fd->name_len == 0 || fd->name_len > 3) {
-		if (fd->name_len == 0 && in->block_ctx && (fd->json_flags & T_VALUE) &&
-		    (fd->json_flags & T_STRING_END)) {
-			return SENML_JSON_STRING_BLOCK_DATA;
-		}
-		return SENML_JSON_UNKNOWN_ATTRIBUTE;
-	}
+struct senml_json_object {
+	union {
+		struct senml_float_payload  float_obj;
+		struct senml_float_bn_payload float_bn_obj;
+		struct senml_boolean_payload boolean_obj;
+		struct senml_boolean_bn_payload boolean_bn_obj;
+		struct senml_opaque_payload  opaque_obj;
+		struct senml_opaque_bn_payload  opaque_bn_obj;
+		struct senml_string_payload string_obj;
+		struct senml_string_bn_payload string_bn_obj;
+	} obj;
+};
 
-	if (buf_read(attrbute_name, fd->name_len, CPKT_BUF_READ(in->in_cpkt), &fd->name_offset) <
-	    0) {
-		LOG_ERR("Error parsing attribute name!");
-		return SENML_JSON_UNKNOWN_ATTRIBUTE;
-	}
+struct json_out_formatter_data {
+	/* flags */
```

Comment:
Do we need those comments? They don't add much value as they simply repeat the struct member name.

Reply:
Ok I remove those.